src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -945,35 +1055,58 @@ struct ScourgeMinion : public CombatAI
 struct PallidHorrorAI : public CombatAI
 {
     std::set<ObjectGuid> m_flameshockers;
+    std::unordered_map<ObjectGuid, uint32> m_flameshockFollowers;
 
     PallidHorrorAI(Creature* creature) : CombatAI(creature, 999)
     {
+        AddCustomAction(25, 2500u, [&]() // Hacky solution for flameshockers to remain in formation
```

Comment:
Very hacky solution to keep the followers in a circle formation. Doesn't really work, too because their walk-speed is too low. I hope the `JoinCreatureGroup` function can take care of this in the future

Reply:
Maybe cyberium formation system cannhelp you now with this